flow_py_sdk/client/entities.py

```diff
@@ -149,17 +150,36 @@ def __init__(
         self.transaction_index: int = transaction_index
         self.event_index: int = event_index
         self.payload: bytes = payload
-        self.value: cadence.Event = json.loads(payload, object_hook=cadence_object_hook)
+
+        try:
+            # Attempt to decode the payload
+            self.value: cadence.Event = json.loads(
+                payload, object_hook=cadence_object_hook
+            )
+        except json.JSONDecodeError as e:
+            logging.error(
+                f"JSON decode error for event {event_index} with payload: {payload[:100]}... Error: {str(e)}"
+            )
+            self.value = None
```

Comment:
I like the addition of logging, but I would not return None, as it can lead to confusing results for users (everything seems ok, but no value is returned). I would instead just re-raise the exception here. n

same goes for all try-catch blocks in this file: `flow_py_sdk/client/entities.py`.